src/lib/components/Uploader/uploader.component.js

```diff
@@ -0,0 +1,236 @@
+import React, { Component } from "react";
+import { lookup, extension } from "mime-types";
+import auth from "solid-auth-client";
+import { UploadedFiles, SolidError } from "@entities";
+
+type Props = {
+  fileBase: String,
+  limitFiles: number,
+  render: Node,
+  onComplete?: (files: Array<UploadedFiles>) => void,
+  onDrop?: (files: Array<UploadedFiles>) => void,
+  onError?: (error: SolidError) => void,
+  onStart?: () => void,
+};
+
+class Uploader extends Component<Props> {
+  counter: number;
+  positionFile: number;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      dragging: false,
+      files: [],
+      uploadedFiles: []
+    };
+    this.fileInput = React.createRef();
+    this.positionFile = 0;
+  }
+
+  componentDidMount() {
+    window.addEventListener("dragover", (event: React.EventHandler) => {
+      this.overrideEvent(event);
+    });
+    window.addEventListener("drop", (event: React.EventHandler) => {
+      this.overrideEvent(event);
+    });
+  }
+
+  componentDidUpdate(prevProps: Object, prevState: Object) {
+    // When files updates means that we have a new files to upload
+    if (this.state.files !== prevState.files && this.state.files.length > 0) {
+      this.upload();
+    }
+    // we check if all files was uploaded and fire onComplete props
```

Comment:
Minor fix, should read 

// We check if all files were uploaded, then fire the onComplete handler

Reply:
done